### PR TITLE
Add emoji support in markup with emoji short name surrounded with colons

### PR DESCRIPTION
**Pull request by YuriyGorvitovskiy**

### CODE REVIEW

**Review by agavrilov — CHANGES_REQUESTED** [review]
Can you use emoji-toolkit as dependency instead of copying emoji.json (https://github.com/joypixels/emoji-toolkit)? It will avoid a duplication of very big file

**Comment by agavrilov** [comment]
@YuriyGorvitovskiy See name2emoji function in RichText.tsx in our code

**Comment by YuriyGorvitovskiy** [comment]
It is not license friendly. Did somebody validate it?
https://www.joypixels.com/licenses/enterprise

**Comment by agavrilov** [comment]
> 
> 
> It is not license friendly. Did somebody validate it?
> https://www.joypixels.com/licenses/enterprise

This license for images only. For the code, it is MIT: https://github.com/joypixels/emoji-toolkit/blob/master/LICENSE.md

**Comment by YuriyGorvitovskiy** [comment]
Changed implementation to use emoji-toolkit

File: index.tsx [hunk 1]
```diff
@@ -395,7 +396,15 @@ const BLOCK_SYNTAXES = [
   NP_TABLE_R,
   PARAGRAPH_R,
 ]
-
+/*
```
Comment (agavrilov) [inline index.tsx]: No need for this code anymore

File: tsconfig.json [hunk 2]
```diff
@@ -9,5 +10,8 @@
     "outDir": "./dist",
     "target": "ES2018"
   },
-  "files": ["index.tsx"]
-}
+  "files": [
+    "index.tsx",
+    "emoji.json"
```
Comment (agavrilov) [inline tsconfig.json]: no emoji.json

File: tsconfig.json [hunk 3]
```diff
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
     "declaration": true,
+    "resolveJsonModule": true,
```
Comment (agavrilov) [inline tsconfig.json]: no need for this option

**Review by agavrilov — CHANGES_REQUESTED** [review]

**Review by agavrilov — APPROVED** [review]

**Pull request merged**